indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -125,10 +124,13 @@ public void run()
         log.info("Started WorkerTaskMonitor.");
         started = true;
       }
+      catch (InterruptedException e) {
+        throw e;
```

Comment:
In this case interrupting should mean "quit now" -- it only occurs when the server is shutting down. Nothing should be logging InterruptedExceptions and keeping going (that was the bug here). They should always fall through and cause stuff to stop.